pkg/ingester/ingester_v2_test.go

```diff
@@ -0,0 +1,288 @@
+				{Labels: client.FromLabelsToLabelAdapters(series[0].lbls), Samples: []client.Sample{{Value: 1, TimestampMs: 100000}}},
+				{Labels: client.FromLabelsToLabelAdapters(series[1].lbls), Samples: []client.Sample{{Value: 1, TimestampMs: 110000}}},
+			},
+		},
+		"should filter series by !~ matcher": {
+			from: math.MinInt64,
+			to:   math.MaxInt64,
+			matchers: []*client.LabelMatcher{
+				{Type: client.REGEX_NO_MATCH, Name: model.MetricNameLabel, Value: ".*_1"},
+			},
+			expected: []client.TimeSeries{
+				{Labels: client.FromLabelsToLabelAdapters(series[2].lbls), Samples: []client.Sample{{Value: 2, TimestampMs: 200000}}},
+			},
+		},
+		"should filter series by multiple matchers": {
+			from: math.MinInt64,
+			to:   math.MaxInt64,
+			matchers: []*client.LabelMatcher{
+				{Type: client.EQUAL, Name: model.MetricNameLabel, Value: "test_1"},
+				{Type: client.REGEX_MATCH, Name: "status", Value: "5.."},
+			},
+			expected: []client.TimeSeries{
+				{Labels: client.FromLabelsToLabelAdapters(series[1].lbls), Samples: []client.Sample{{Value: 1, TimestampMs: 110000}}},
+			},
+		},
+		"should filter series by matcher and time range": {
+			from: 100000,
+			to:   100000,
+			matchers: []*client.LabelMatcher{
+				{Type: client.EQUAL, Name: model.MetricNameLabel, Value: "test_1"},
+			},
+			expected: []client.TimeSeries{
+				{Labels: client.FromLabelsToLabelAdapters(series[0].lbls), Samples: []client.Sample{{Value: 1, TimestampMs: 100000}}},
+			},
+		},
+	}
+
+	// Create ingester
+	i, cleanup, err := newIngesterMockWithTSDBStorage(defaultIngesterTestConfig())
+	require.NoError(t, err)
+	defer i.Shutdown()
+	defer cleanup()
+
+	// Wait until it's ACTIVE
+	test.Poll(t, 1*time.Second, ring.ACTIVE, func() interface{} {
+		return i.lifecycler.GetState()
+	})
+
+	// Push series
+	ctx := user.InjectOrgID(context.Background(), "test")
+
+	for _, series := range series {
+		req, _ := mockWriteRequest(series.lbls, series.value, series.timestamp)
+		_, err := i.v2Push(ctx, req)
+		require.NoError(t, err)
+	}
+
+	// Get label values
+	for testName, testData := range tests {
+		t.Run(testName, func(t *testing.T) {
+			req := &client.QueryRequest{
+				StartTimestampMs: testData.from,
+				EndTimestampMs:   testData.to,
+				Matchers:         testData.matchers,
+			}
+
+			res, err := i.v2Query(ctx, req)
+			require.NoError(t, err)
+			assert.ElementsMatch(t, testData.expected, res.Timeseries)
+		})
+	}
+}
+
+func mockWriteRequest(lbls labels.Labels, value float64, timestampMs int64) (*client.WriteRequest, *client.QueryResponse) {
+	samples := []client.Sample{
+		{
+			TimestampMs: timestampMs,
+			Value:       value,
+		},
+	}
+
+	req := client.ToWriteRequest([]labels.Labels{lbls}, samples, client.API)
+
+	// Generate the expected response
+	expectedResponse := &client.QueryResponse{
+		Timeseries: []client.TimeSeries{
+			{
+				Labels:  client.FromLabelsToLabelAdapters(lbls),
+				Samples: samples,
+			},
+		},
+	}
+
+	return req, expectedResponse
```

Comment:
I don't see `expectedResponse` being used anywhere

Reply:
This function is a copy and paste from https://github.com/cortexproject/cortex/pull/1830 (still under review). Keeping the copy and paste will help me rebasing once one of the two PRs will get merged.